receiver/hostmetricsreceiver/README.md

```diff
@@ -143,6 +143,48 @@ service:
 
 ### Feature gate configurations
 
+#### Transition to process memory metric names aligned with OpenTelemetry specification
+
+The Host Metrics receiver has been emitting the following process memory metrics:
+
+- [process.memory.physical_usage] for the amount of physical memory used by the process,
+- [process.memory.virtual_usage] for the amount of virtual memory used by the process.
+
+This is in conflict with the OpenTelemetry specification,
+which defines [process.memory.usage] and [process.memory.virtual] as the names for these metrics.
+
+To align the emitted metric names with the OpenTelemetry specification,
+the following feature gates have been introduced:
+
+- `receiver.hostmetricsreceiver.emitCorrectProcessMemoryMetrics`,
+- `receiver.hostmetricsreceiver.emitIncorrectProcessMemoryMetrics`,
```

Comment:
Do we really need to control it with feature gates? I believe we can use the `enabled` config options instead...
1. add new metrics as disabled by default, show warning if old are used
2. make old metrics disabled by default, enable new ones
3. remove old metrics

WDYT?

I would also recommend making the transition process longer. Like 3 releases for each step to have enough time for users and backends to adopt for the new changes

Reply:
> Do we really need to control it with feature gates?

Yes, functionally the same thing can be achieved just with the regular otelcol config. The difference is in the UX - the feature gates can be operated in CLI only, while the otelcol config can be specified in the config file and via CLI (using the `--set` argument). So I guess there is no downside in using the config instead of feature gates? Other than perhaps discoverability, encapsulation?... I'm happy to rework the solution to use config instead of feature gates, if that's the decision. Please let me know @dmitryax.

> I would also recommend making the transition process longer. 

I totally agree, given we release every 2 weeks, the timeline I initially proposed is very aggressive. I'm all for making the transition process longer. Three releases are fine by me. Will adjust the documentation.

Reply:
I've added a commit that extends the transition timeline, giving three releases to each step.

Reply:
Controlling it in the config seems more flexible to me. Not sure I understand the discoverability and encapsulation concerns. Do you think config options will be less discoverable if we document them the same way as the feature gates? Also, if we control them in the config, we can have them both rendered in the metrics list (documentation.md) with old ones marked as deprecated.

Reply:
Sure, I'll remove the feature gates and describe the use of the config to manage the transition. :+1:

Reply:
@astencel-sumo after taking another look, I see a benefit of feature gates :) it's two feature gates vs 4 config options in total. So maybe we can keep feature gates, if you think that easier option for users to follow. Let me know

Reply:
:smile: How about we bring this subject to the Collector SIG meeting today?

Reply:
As discussed during the SIG, there is no preference among other interested people. I probably still slightly prefer `enabled` options over the feature gates, but I'm good with any approach. Please let me know which one you prefer and go ahead with that.

Also I tried to come up with some rules for the metric receivers that I would like us to follow going forward https://github.com/open-telemetry/opentelemetry-collector/pull/6358 Feel free to share your feedback

Reply:
After thinking on it, I would also lean toward using configuration file options. In my experience the typical way to configure the receiver and the Collector as a whole is through the configuration file and not through passing arguments to the executable, so toggling `enabled` options may be simpler for users to do. I also believe that the Collector is moving toward preferring the configuration file to command line arguments in general. I'm not sure about when other components use feature gates though and am not familiar with how others deploy and configure the Collector, so I could be off base here.

Reply:
:100: I agree with the your points Evan

One more argument in favor of config files is why introduce more code (for the feature gates) when there is already code that can do what you want to achieve? The less code the better :smile: